### ISSUE TITLE
watch option not working?

### ISSUE THREAD

**Issue by yss14** [sunknudsen/npm-hardlink#1]
Thanks for this great tool!

Works like a charm, except the `--watch` option.
The console output stays at 
```
...
Dependencies installed
Watching react-safeway for changes…
```
even if the package is recompiled by tsc and the new output files are present in the `build/` folder of my package.

My package.json of the local package looks like
```json
{
	"name": "<name>",
	"version": "1.0.0",
	"description": "",
	"main": "dist/index.js",
        ....
}
```

System: macOS 11.1 (Apple Silicon)

**Comment by sunknudsen** [comment]
Hey Yannick, thanks for reporting the issue. Is the destination updated? Are you using npm-hardlink in the context of a CRA project?

**Comment by sunknudsen** [comment]
@yss14 Btw, the `--watch` option does not output anything on file changes (even though it created hard links or copies files in the background). perhaps this is what you are experiencing.

**Comment by yss14** [comment]
> Are you using npm-hardlink in the context of a CRA project?
Yes!

> Btw, the --watch option does not output anything on file changes (even though it creates hard links or copies files in the background). perhaps this is what you are experiencing.

Ok, so for new files I have to run the `npm-hardlink` command again, right?

**Comment by sunknudsen** [comment]
@yss14 Is the issue that your CRA app is not compiling and the browser is not reloading when you edit files in the source while using the `--watch` option?

If so, what version of `npm-hardlink` are you using (`npm ls -g npm-hardlink`)? I just published version `0.0.6` which includes a new `--copy` option that copies files vs using hard links (which I believe is broken in [chokidar](https://github.com/paulmillr/chokidar), see https://github.com/paulmillr/chokidar/issues/1071).

**Comment by yss14** [comment]
> Is the issue that your CRA app is not compiling and the browser is not reloading when you edit files in the source while using the --watch option?

Yes, I tried both, hard links and copy, but both don't trigger a recompilation.

**Comment by sunknudsen** [comment]
When you run `npm-hardlink --copy ...` are the files that should be copied to the destination listed?

If they are, you are likely hitting an issue that I also have on Catalina (see https://github.com/paulmillr/chokidar/issues/1071 and https://github.com/facebook/create-react-app/issues/10475).

**Comment by yss14** [comment]
Yes, when I run `npx npm-hardlink --dest <path> --install-deps-prod --watch --copy` all files of the `dist` folder are listed.

So should the intended behaviour be that on a file change should trigger a "re-copy" of that file to the destination?

**Comment by yss14** [comment]
I just came up with a little (nasty) workaround as long as the issue with webpack persists.
It's a little bit slow, but seems to work.

```javascript
const fs = require("fs")
const spawn = require("child_process").spawn

let hardLinkProcess = null

fs.watch("./dist", () => {
	if (hardLinkProcess) {
		hardLinkProcess.kill("SIGINT")
	}

	hardLinkProcess = spawn(
		"npm-hardlink",
		["--dest", "<path>", "--install-deps-prod", "--copy", "--yes"],
		{ env: process.env },
	)
	hardLinkProcess.stderr.pipe(process.stdout)
	hardLinkProcess.stdout.pipe(process.stdout)
})
```

**Comment by sunknudsen** [comment]
> Edit: The file changed is successfully copied to destination. So the issue really is paulmillr/chokidar#1071. Thanks for helping me out!

Thanks for sharing

> I just came up with a little (nasty) workaround as long as the issue with webpack persists.

Thanks for working on a workaround. Feels good to know I'm not alone tacking this issue.

I don't fully understand how your workaround works... does the above snippet somehow trigger a CRA/Webpack/chokidar watch event?

Btw, I am currently refactoring npm-hardlink to drop hard links and (hopefully) use rsync to speed things up considerably.

The project will likely be renamed `npm-sync`. I will keep you in the loop as soon as I have a beta release.

**Comment by yss14** [comment]
> I don't fully understand how your workaround works... does the above snippet somehow trigger a CRA/Webpack/chokidar watch event?

Me neither, I just discovered that calling `npm-hardlink` again triggers a webpack refresh.

**Comment by sunknudsen** [comment]
Hey,

I just published [npm-sync](https://www.npmjs.com/package/npm-sync) and `npm-hardlink` has been deprecated.

I noticed that hard links are not watched by [chokidar](https://www.npmjs.com/package/chokidar) so the package name didn't work anymore.

The API is very similar but, under the hood, `rsync` is now used for the initial sync (which is significantly more efficient when running `npm-sync` more than once).

Also, `--install-deps` and `--install-deps-prod` no longer exists as syncing package `node_modules` is the default (it can be disabled using `--no-deps`.

This change in dependency behavior is done to support recursive use cases like the following.

Project `sunknudsen-website` depends on `@sunknudsen/react-cms` which depends on `@sunknudsen/react-hash-links` and both `@sunknudsen/react-cms` and `@sunknudsen/react-hash-links` are in dev mode (meaning both are running `npm -sync`).

Running `npm-sync --dest /Users/sunknudsen/Code/sunknudsen/sunknudsen-website --watch` in `@sunknudsen/react-cms` and `npm-sync --dest /Users/sunknudsen/Code/sunknudsen/sunknudsen-website --watch` in `@sunknudsen/react-hash-links` triggers watch events all the way down to `sunknudsen-website`.

Let me know if you run into issues (aside from the CRA/Webpack/chokidar inconsistencies).

**Comment by sunknudsen** [comment]
@yss14 This repo will de deleted shortly. Looking forward to continuing our conversation on https://github.com/sunknudsen/npm-sync.

**Comment by sunknudsen** [comment]
@yss14 I believe the issue is caused by fsevents (see [this](https://github.com/fsevents/fsevents/issues/355) issue).

Do you have time to try to reproduce https://github.com/fsevents/fsevents/issues/355#issuecomment-770385182 on Big Sur on your M1 hardware?

**Comment by yss14** [comment]
@sunknudsen I followed your instructions and tried to reproduce the issue with your [demo repo](https://github.com/sunknudsen/fsevents-test).
After saving the file 30 times the error occured!

```
Assertion failed: (napi_create_string_utf8(env, events->events[idx].path, NAPI_AUTO_LENGTH, &args[0]) == napi_ok), function fse_dispatch_event, file ../src/fsevents.c, line 147.
zsh: abort      node index.js
```

**Comment by sunknudsen** [comment]
Thanks for sharing @yss14. How many times did you run the test? On my computer, the error is thrown after 2-4 edits/saves.

Would you mind sharing the results of your runs on https://github.com/fsevents/fsevents/issues/355?
